socorro/external/pubsub/crashqueue.py

```diff
@@ -128,11 +129,17 @@ def __iter__(self):
         while True:
             msgs = 0
             for sub_path in sub_paths:
-                response = self.subscriber.pull(
-                    sub_path,
-                    max_messages=PUBSUB_MAX_MESSAGES,
-                    return_immediately=True
-                )
+                try:
+                    response = self.subscriber.pull(
+                        sub_path,
+                        max_messages=PUBSUB_MAX_MESSAGES,
+                        return_immediately=True
+                    )
+                except DeadlineExceeded:
+                    # If we exceeded the timeout, treat it as if there was
+                    # nothing to receive and move on.
+                    continue
```

Comment:
I think this is the best thing to do. It might be the case that we hit this a lot and we wouldn't know because we're silently ignoring it, but given that we've only hit a couple of these in the month the code has been live, I don't think that will be the case.